### PR TITLE
insomnia-designer: init at 2020.4.1

### PR DESCRIPTION
**Pull request by Pablo1107**
###### Motivation for this change
Add `insomnia-designer`.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by puzzlewolf** [comment]
Hey, any particular reason this is built from a `.deb` package and not from source?

File: pkgs/development/web/insomnia/designer.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
+  pname = "insomnia-designer";
+  version = "2020.2.2";
+
+  src = fetchurl {
+    url =
+      "https://github.com/Kong/insomnia/releases/download/designer@${version}/Insomnia.Designer-${version}.deb";
```
Comment (Sohalt) [inline pkgs/development/web/insomnia/designer.nix:23]: Is there a reason why you are using the debian package instead of building from source?

Comment (Pablo1107) [reply pkgs/development/web/insomnia/designer.nix:23]: I take the same approach as the currently available insomnia. I imagine is feasible to build it from source but I didn't try it and this worked well enough for me.

**Comment by Pablo1107** [comment]
> Hey, any particular reason this is built from a `.deb` package and not from source?

As I answer to @Sohalt : I took the same approach as the currently available insomnia. I imagine is feasible to build it from source but I didn't try it and this worked well enough for me.

**Comment by Pablo1107** [comment]
@Sohalt @puzzlewolf will building from source be a requirement for this package to be merged?

I wonder how `insomnia` got merged if this is the case.

**Review by puzzlewolf — CHANGES_REQUESTED** [review]
No, building from source is not required, but it's usually preferred.

I don't have much experience with this kind of package, sry. But the the derivation is very similar to `insomnia`, and that got merged. So I guess this is ok :)

Result of `nixpkgs-review pr 92093` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- insomnia-designer
</details>

File: pkgs/development/web/insomnia/designer.nix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  preFixup = ''
+    wrapProgram "$out/bin/designer" --prefix LD_LIBRARY_PATH : ${runtimeLibs}
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://insomnia.rest/";
+    description = "The most intuitive cross-platform REST API Client";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ markus1189 babariviere ];
```
[the remark this answers is not in the document]
Comment (Pablo1107) [reply pkgs/development/web/insomnia/designer.nix:89]: Oh, I completely forgot to add myself as maintainer, will correct it.

**Comment by Pablo1107** [comment]
@puzzlewolf I added myself as maintainer and fixed metadata.

**Review by puzzlewolf — APPROVED** [review]
looks good to me now :+1:

**Comment by Pablo1107** [comment]
Update to 2020.4.1. I've been successfully running this package since several months.

**Review by babariviere — APPROVED** [review]

File: pkgs/development/web/insomnia/designer.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
+  pname = "insomnia-designer";
+  version = "2020.4.1";
+
+  src = fetchurl {
+    url =
+      "https://github.com/Kong/insomnia/releases/download/designer@${version}/Insomnia.Designer-${version}.deb";
```
Comment (SuperSandro2000) [inline pkgs/development/web/insomnia/designer.nix:23]:
```suggestion
    url = "https://github.com/Kong/insomnia/releases/download/designer@${version}/Insomnia.Designer-${version}.deb";
```

File: pkgs/development/web/insomnia/designer.nix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    sha256 = "06jarli3n6anx3hqdwvs1j9cim0dfw59zvhdf6dhp8kgps7z3aq7";
+  };
+
+  nativeBuildInputs =
+    [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/web/insomnia/designer.nix:28]:
```suggestion
  nativeBuildInputs = [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```

File: pkgs/development/web/insomnia/designer.nix [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://insomnia.rest/";
+    description = "The most intuitive cross-platform REST API Client";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ markus1189 babariviere ];
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/web/insomnia/designer.nix:91]:
```suggestion
  };
```

File: pkgs/development/web/insomnia/designer.nix [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  preFixup = ''
+    wrapProgram "$out/bin/insomnia-designer" --prefix LD_LIBRARY_PATH : ${runtimeLibs}
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://insomnia.rest/";
+    description = "The most intuitive cross-platform REST API Client";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ markus1189 babariviere ];
```
Comment (SuperSandro2000) [inline pkgs/development/web/insomnia/designer.nix:89]: Please add yourself as maintainer and not other people.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 92093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>insomnia-designer</li>
  </ul>
</details>

**Comment by Ekleog** [comment]
@Pablo1107 Would you be willing to implement @SuperSandro2000's suggestions? I have not reviewed myself yet, but this discussion looks to me like when you implement them it'll become possible to land this PR, and hopefully the suggestions should be simple to implement :) (I'd just go ahead, click “Commit suggestion” and land myself right now, but I don't want to add you as a maintainer if you don't actually want to be :/)

**Comment by Pablo1107** [comment]
@Ekleog yes, I'll see to push those changes so you can merge it.

**Pull request closed without merging**